app/jobs/fetch_resources_job.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Load JSON resources from a url, persist them, and index them
+class FetchResourcesJob < ApplicationJob
+  queue_as :default
+
+  def perform(url, exhibit)
+    resp = Faraday.get(url)
+
+    resource = resp.body.split("\n")
+    resource.each_with_index do |item, index|
+      create_or_update_resource(item, exhibit, index)
+    end
+    logger.info("#{resource.count} records were created from #{url}.")
+  end
+
+  private
+
+  def create_or_update_resource(item, exhibit, index)
+    json = JSON.parse(item)
+    resource = DlmeJson.find_or_initialize_by(url: json['id'], exhibit: exhibit)
+    resource.data = { json: item }
+
+    raise "Resource #{index + 1} in #{url} is not valid." unless resource.valid?
+
+    resource.save_and_index
```

Comment:
`save_and_index` is an async operation. Since this is a background job, maybe it should be synchronous here? See https://github.com/projectblacklight/spotlight/blob/3b30e827ba40772d519ec5123f5b8c7dd945131f/app/models/spotlight/resource.rb#L27-L29